resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -806,11 +796,31 @@ private[yarn] class YarnAllocator(
     }
   }
 
+  // visible for testing
```

Comment:
nit. Could you try to use `PrivateMethodTester`? Here is an example, cfd42355ccbdf34ca9bd5a125c4310b9da14100c